admin-dev/themes/new-theme/js/pages/feature-flag/index.js

```diff
@@ -0,0 +1,51 @@
+/**
+ * Copyright since 2007 PrestaShop SA and Contributors
+ * PrestaShop is an International Registered Trademark & Property of PrestaShop SA
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.md.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to https://devdocs.prestashop.com/ for more information.
+ *
+ * @author    PrestaShop SA and Contributors <contact@prestashop.com>
+ * @copyright Since 2007 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ */
+
+import ConfirmModal from '@components/modal';
+
+const {$} = window;
+
+$(() => {
+  const $submitButton = $('#submit-btn-feature-flag');
+  const $form = $('#feature-flag-form');
+
+  $submitButton.on('click', (event) => {
```

Comment:
Rather than check the button click, it's safer to check the form submit event
The form can be submitted via the keyboard (it you use tab navigation it's doable even without text inputs I think)

Reply:
I agree with this, if the button is selectable by a tabulation, that would cause issue, but maybe it's not possible in this case

Reply:
@jolelievre if I replace
```
$submitButton.on('click', (event) => {
```
by
```
$form.on('submit', (event) => {
```
what do I do when the confirmation is given?

Because my implementation does
```
$form.submit();
```
so right now it's a forever-loop 😆  .

If I listen to form submit to open a modal, I cannot submit the form following the user confirmation input 😋  .

I could manually craft an HTTP request to send but then I lose all the benefits of using a form 🤷‍♂️  .

I'm open to suggestions 😄

Reply:
The way to do this usually is to return false in the handler to block the submit So in your function you can use a `confirm` or a modal And if the user clicks on no return false

If you use our modal I think you will need to use an `async` function to display it This way you can use `await` before it to make it synchronuous

Reply:
@NeOMakinG It seems I cannot easily add `async` as the Modal component is a stand-alone component used in many pages.
I could create functions "just to add async in front" but isn't it overkill? Can you tell me how you would do it?

Here is what I want to achieve
```
$(() => {
  const $submitButton = $('#submit-btn-feature-flag');
  const $form = $('#feature-flag-form');

  $form.on('submit', (event) => {
    event.preventDefault();

    const modal = new ConfirmModal(
      {
        id: 'modal-confirm-submit-feature-flag',
        confirmTitle: $submitButton.data('modal-title'),
        confirmMessage: $submitButton.data('modal-message'),
        confirmButtonLabel: $submitButton.data('modal-apply'),
        closeButtonLabel: $submitButton.data('modal-cancel'),
      },
      () => {
        return true; // this is the success callback
      },
      () => {
        return false; // this is the cancel callback
      },
    );
    await modal.show(); // I need to await here, right? and then return the result
  });
});
```

Reply:
I didn't think about all this Maybe to allow this it requires to transform `modal.show()` into an async function 🤔

Or the constructor or something else, but I think it would be a great improvement, because having a synchronous confirm modal is a very common use case, and if our custom modal doesn't allow it It means we can't substitute it to a native `confirm` method